wolfcrypt/src/sha.c

```diff
@@ -356,8 +369,11 @@ static INLINE void AddLength(Sha* sha, word32 len)
 }
 
 
-int wc_ShaUpdate(Sha* sha, const byte* data, word32 len)
+int wc_ShaUpdate (Sha* sha, const byte* data, word32 len)
 {
+    if (sha == NULL ||(data == NULL && len > 0)) {
+        return BAD_FUNC_ARG;
+    }
     /* do block size increments */
     byte* local = (byte*)sha->buffer;
```

Comment:
move variable declarations to beginning of statement or change to "declaration then assignement"